src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +148,221 @@ async function uploadWithProgress({
+            canPickMany: false,
+            ignoreFocusOut: true,
+            title: localize('AWS.message.selectBucket','Select S3 bucket to upload to'),//localize('AWS.lambda.upload.title', 'Select Upload Type'),
+            step: 2,
+            totalSteps: 2,
+        },
+        items: [...bucketItems, createNewBucket],
+        buttons: [vscode.QuickInputButtons.Back]
+    })
+    const response = verifySinglePickerOutput(await promptUser({
+        picker: picker,
+        onDidTriggerButton: (button, resolve, reject) => {
+            if (button === vscode.QuickInputButtons.Back) {
+                resolve([{
+                    label: "back",
+                    bucket: undefined
+                }])
+            }
+
+        },
+    })) 
+
+    if (!response) {
+        return undefined
+    }
+
+    if (!response.bucket) {
+        if (response.label === 'back') {
+            uploadFilePaletteCommand()
+        }
+        if (response.label === 'Create new bucket') {
+            getLogger().debug('CreateBucket called for: %O', s3client)
+
+            const bucketName = await window.showInputBox({
+                prompt: localize('AWS.s3.createBucket.prompt', 'Enter a new bucket name'),
+                placeHolder: localize('AWS.s3.createBucket.placeHolder', 'Bucket Name'),
+                validateInput: validateBucketName,
+            })
+
+            if (!bucketName) {
+                getLogger().info('CreateBucket cancelled')
+                telemetry.recordS3CreateBucket({ result: 'Cancelled' })
+                return undefined
+            }
+
+            getLogger().info(`Creating bucket: ${bucketName}`)
+            try {
+                const bucket = await s3client.createBucket({ bucketName })
+
+                getLogger().info('Created bucket: %O', bucket)
+                window.showInformationMessage(localize('AWS.s3.createBucket.success', 'Created bucket: {0}', bucketName))
+                telemetry.recordS3CreateBucket({ result: 'Succeeded' })
+                return { Name: bucket.bucket.name }
+            } catch (e) {
+                getLogger().error(`Failed to create bucket ${bucketName}: %O`, e)
+                showErrorWithLogs(
+                    localize('AWS.s3.createBucket.error.general', 'Failed to create bucket: {0}', bucketName),
+                    Window.vscode()
+                )
+                telemetry.recordS3CreateBucket({ result: 'Failed' })
+                return undefined
+            }
+        }
+    } else {
+        return response.bucket
+    }
+    return undefined
+}
+
+export async function uploadFilePaletteCommand(
+    fileSizeBytes: FileSizeBytes = statFile,
+    window = Window.vscode(),
+    outputChannel = ext.outputChannel
+): Promise<void>{
+
+    //1 step: get the file that is currently opened in the editor
+    const activeEditor: TextEditor | undefined = vscode.window.activeTextEditor // returns an object of type TextEditor | undef 
```

Comment:
Unnecessary comment at end (typing information is already present in const definition, in fact, typing information isn't required since it will be inferred from the assignment).

Consider: elevate this logic to the command registration and pass the file location in as a parameter. That way, you'll be able to use this functionality elsewhere (you may want to rename the function to make it not Command Palette-specific).

Reply:
`uploadToS3Command` work?